llmfoundry/utils/warnings.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+
+class VersionedDeprecationWarning(DeprecationWarning):
```

Comment:
Should we upstream this to composer?

Reply:
hmm could but then we'd need to wait for a release to use it in foundry?

Reply:
yeah you can also just do it in this pr and open a companion PR to composer